bookwyrm/static/css/bookwyrm.css

```diff
@@ -202,25 +194,85 @@ html {
     min-height: 96px;
 }
 
-/* --- QUOTES --- */
-.quote blockquote {
+/** Statuses: Quotes
+ *
+ * \e906: icon-quote-open
+ * \e905: icon-quote-close
+ *
+ * The `content` class on the blockquote allows to apply styles to markdown
+ * generated HTML in the quote: https://bulma.io/documentation/elements/content/
+ *
+ * ```html
+ * <div class="quote block">
+ *     <blockquote dir="auto" class="content mb-2">
+ *         User generated quote in markdown…
+ *     </blockquote>
+ *
+ *     <p> — <a…>Book Title</a> by <a…class="author">Author</a></p>
+ * </div>
+ * ```
+ ******************************************************************************/
+
+.quote > blockquote {
     position: relative;
     padding-left: 2em;
 }
 
-.quote blockquote::before,
-.quote blockquote::after {
+.quote > blockquote::before,
+.quote > blockquote::after {
     font-family: 'icomoon';
     position: absolute;
 }
 
-.quote blockquote::before {
+.quote > blockquote::before {
     content: "\e906";
     top: 0;
     left: 0;
 }
 
-.quote blockquote::after {
+.quote > blockquote::after {
     content: "\e905";
     right: 0;
 }
+
+/** Animations and transitions
+ ******************************************************************************/
+
+@keyframes beating {
+    0% { font-size: 100%; }
+    25% { font-size: 125%; }
+    100% { font-size: 75%; }
+}
+
+@keyframes turning {
+    from { transform: rotateZ(0deg); }
+    to { transform: rotateZ(360deg); }
+}
+
+.is-processing .icon-heart::before {
+    animation: beating 0.5s infinite alternate ease-in-out;
+}
+
+.is-processing .icon-boost::before {
+    animation: turning 0.5s infinite ease-in-out;
+}
```

Comment:
Let me give more details about this specific intent which I hope could be summarised by [How well does it fail](https://adactio.com/articles/12839).

The idea here is to have some kind of _loading_ animation, so having just one pulse/turn/whatever would defeat that purpose. Not all requests are made in good network conditions, which means they can take a long time or fail if packets are lost, if there is suddenly no network, if the user is unaware they got disconnected from their session while still looking at the same UI, etc.

Let’s test about the speed and breath of scaling.

Reply:
I quickly played around with values and the result was not visible enough at such a small scale or, frankly, a bit stressful if pulsing faster. If you have a quantifiable proposal to make, I will gladly update accordingly or welcome a commit.

Reply:
I hadn't looked into _why_ it was lasting for so long when I initially commented, so I didn't realize it was intended as a loading animation rather than an interaction feedback. The visual result doesn't suggest to me that it's a loading state, and as with follow buttons, waiting for the http response isn't a pattern that reflects how the application works. 

I'd be happy to include a very small animation on transition at some point, but for the moment, the multi-step `is-processing` class model that waits for the server response to update isn't a pattern that works with the application.

Reply:
Sorry… I’m confused with this part:

> waiting for the http response isn't a pattern that reflects how the application works.

What would be the pattern that reflects how the application works?

As implemented in this PR now:

1. The user asks for something (boost, favourite, follow, unfollow).
1. The button used to submit the form is disabled while the request is made, with or without JS.
1. The state of the button and the form is updated accordingly to the response received by the server, by JS or by the updated page.

I’m not entirely sure what is not reflecting how the application is working since what is seen is actually what happens, and prevent multiple submissions. :thinking: 

I will remove the animation, no worries. Shall I remove the hook (toggling the `is-processing` class), or keep it so that it will be there to use if a better visual solution is agreed upon.

Reply:
There are a few reasons to show the undo version of the actions immediately rather than waiting: The initial http response doesn't always reflect the actual outcome of the action, because the success may be dependent on confirmation from a remote instance, and the whole transaction can take many minutes even when operating normally. Also the application's performance isn't great right now. And, what I think is the most compelling reason, is that an undo action (unfollow, un-boost, etc) is functional to the user immediately - for example, if I follow a user, unfollowing them immediately after, even if somehow the follow hasn't saved in the database yet, will still successfully cancel that follow that I sent out. And if the follow fails, the unfollow will at worst do nothing, and at best clarify the application state.

Reply:
I see. Thanks for taking the time to explain further; I understand.

I updated the code so that the buttons are swapped immediately as they were. I kept the Promise handling as it does not hurt anyone.

Reply:
I have high hopes for the promise handing being very useful one day :)